sympy/matrices/matrices.py

```diff
@@ -189,6 +189,9 @@ def __neg__(self):
         return -1*self
 
     def _matrix_pow_by_jordan_blocks(self, num):
+        if (num < 0) == True and self.det() == 0:
```

Comment:
>If the determinant is computed, I think it could be done before calling _matrix_pow_by_jordan_blocks

If det computed and raised in ```__pow__```, it won't work if we directly call ```_matrix_pow_by_jordan_blocks```.